go/mysql/replication_status.go

```diff
@@ -123,16 +123,7 @@ func (s *ReplicationStatus) FindErrantGTIDs(otherReplicaStatuses []*ReplicationS
 		if !ok {
 			panic("The receiver ReplicationStatus contained a Mysql56GTIDSet in its relay log, but a replica's ReplicationStatus is of another flavor. This should never happen.")
 		}
-		// Copy and throw out primary SID from consideration, so we don't mutate input.
-		otherSetNoPrimarySID := make(Mysql56GTIDSet, len(otherSet))
-		for sid, intervals := range otherSet {
-			if sid == status.SourceUUID {
-				continue
-			}
-			otherSetNoPrimarySID[sid] = intervals
-		}
-
-		otherSets = append(otherSets, otherSetNoPrimarySID)
+		otherSets = append(otherSets, otherSet)
```

Comment:
This change was required because I saw that we had a bug in this code, were even if two servers had the exact same GTID set but had different sources set, we were flagging them both as having errant GTIDs. I have added a test for that in `TestFindErrantGTIDs`. We do not need to remove the GTIDs originating from the source for the servers that we are checking against. We should only remove them for the server that is being tested for errant GTIDs.